nix-tools/flake.nix

```diff
@@ -1,8 +1,19 @@
 {
-  inputs.nixpkgs.follows = "haskellNix/nixpkgs";
-  inputs.haskellNix.url = "github:input-output-hk/haskell.nix";
+  inputs = {
```

Comment:
Changes to flake.nix:
- Added CHaP and iohkNix inputs for building nix-tools statically
- Added a `static-nix-tools-outputs` outputs (mostly for testing debugging, to be removed)
- Added the static nix-tools builds + the zipped builds to `hydraJobs`.

Reply:
> Added CHaP and iohkNix inputs for building nix-tools statically

Do we know what bits of CHaP and iohkNix we need?  It would be nice if we could make static executables without them.

Reply:
I also can't see why this could possibly need CHaP

Reply:
I;ll remove CHaP. 
As for iohkNix, we need it for the static libs, see [here](https://github.com/input-output-hk/haskell.nix/pull/2157/files#diff-73893305286f836b6807e866c710fd769a1032081d9529a9dd7246e946df6255R67)

Reply:
We need them for those static libraries _for packages that use them_! `nix-tools` does not use any of those libraries!